server/models/availability.js

```diff
@@ -0,0 +1,54 @@
+const mongoose = require("mongoose");
+
+
+const dayInformation={
+    isAvaliable:{
+        type:Boolean,
+        default:'true'
+    },
+    startDate:{
+        type:Number,
+        min:0,
+        max:24,
+        required:isAvaliableForDay
+    },
+    endDate:{
+        type:Number,
+        min:0,
+        max:24,
+        required:isAvaliableForDay,
+        validate: [dateValidator, 'Start Date must be less than End Date']
+    }
+}
+function isAvaliableForDay(){
+    if(this.isAvaliable > -1){  
+        return true;
+    }
+    return false;
+}
+
+const availabilitySchema = new mongoose.Schema({
+  petSitterId: {
+    type: mongoose.Schema.Types.ObjectId,
+    required: true,
+    ref: 'Profile'
+  },
+  name:{
+      type:'string',
+      required:'true'
+  },
+  days: {
+  sunday:dayInformation,
+  monday:dayInformation,
+  tuesday:dayInformation,
+  wednesday:dayInformation,
+  thursday:dayInformation,
+  friday:dayInformation,
+  saturday:dayInformation
+  },
+  
+});
+function dateValidator(value) {
+    return this.startDate <= value;
+};
```

Comment:
I'll start by saying this is probably better defined inline.

If we want to refactor this outside of the schema, then it makes more sense to make it both more single-purpose and reusable:
```
function deltaPositiveDateRange(start, end) {
   return end > start;
}
```
<sub>I got creative with the name, you can name it as you like 😎 </sub>

Also, it doesn't make much sense to allow `start === end` (optionally, we can extend this idea: what's the minimum time-slot duration we want to accept?)

Reply:
Done the changes.